### PR TITLE
Add safe string subscript.

### PR DESCRIPTION
**Pull request by nickmshelley**

### CODE REVIEW

**Comment by stephanheilner** [comment]
:+1: from me, but @hiltonc needs to merge it.

File: SwiftificationTests/StringTests.swift [hunk 1]
```diff
@@ -81,4 +81,10 @@ class StringTests: XCTestCase {
         }
     }
     
+    func testSafeSubscript() {
+        let str = "Hi"
+        XCTAssertEqual(str[str.startIndex.advancedBy(1)], str[safe: str.startIndex.advancedBy(1)]!)
```
Comment (hiltonc) [inline SwiftificationTests/StringTests.swift]: !

Comment (nickmshelley) [inline SwiftificationTests/StringTests.swift]: I thought `!` was kosher in tests.

Comment (stephanheilner) [inline SwiftificationTests/StringTests.swift]: I almost commented on this too at first, ten thought, "oh, it's a test, so it's fine". :)

Comment (hiltonc) [inline SwiftificationTests/StringTests.swift]: It's OK for mere mortals, but we hold @nickmshelley to a higher standard.

Comment (stephanheilner) [inline SwiftificationTests/StringTests.swift]: Oh, I forgot to take that into consideration. Good point.

**Pull request merged**